### ISSUE TITLE
`Module.get_attribute/3` returns default when the attribute is set to `nil`

### ISSUE THREAD

**Issue by zachallaun** [elixir-lang/elixir#12610]
### Elixir and Erlang/OTP versions

`main` branch

### Operating system

All.

### Current behavior

```elixir
defmodule DefaultAttributes do
  @attribute nil
  Module.get_attribute(__MODULE__, :attribute, :default) #=> :default
end
```

### Expected behavior

`Module.get_attribute/3` should return `nil` if the attribute has been explicitly set.

---

I noticed this while working on a PR for `Module.get_last_attribute/3` (#12608 for context). I believe it's an easy fix if this is indeed a bug and not intended/undocumented behavior. Alternatively, the behavior could be documented and tested.

**Comment by josevalim** [comment]
This is expected. Maps and keywords work in a similar way. A PR to improve the docs is definitely worth it. :)

**Comment by sabiwara** [comment]
> This is expected. Maps and keywords work in a similar way. A PR to improve the docs is definitely worth it. :)

Unless I misunderstood, I don't think this is the case? Seems like we should return `nil` when set to `nil` if we want to be consistent, not the default?

```elixir
iex(2)> Map.get(%{a: nil}, :a, :default)
nil
iex(3)> Keyword.get([a: nil], :a, :default)
nil
```

**Comment by josevalim** [comment]
You are right, I completely got it backwards.

**Comment by zachallaun** [comment]
Thanks for the confirmation @sabiwara!

I’ll add a fix for this in #12611.

**Comment by josevalim** [comment]
Closing in favor of #12611.
